### DISCUSSION TITLE
Open file on drop

### DISCUSSION THREAD

**Discussion by bulutfatih** [coteditor/CotEditor#1204] [Q&A] [answered]
Is there any way to open a file with drag and drop? It inserts the file path to the current document, instead of opening it.

**Comment by 1024jp** [comment, marked as answer]
Unfortunately, CotEditor does not provite such a feature intentionally at present. Only the area to open a file is the application icon in the Dock or Finder since this is the standard behavior on a document-based macOS application.

**Reply by bulutfatih** [reply]
> Unfortunately, CotEditor does not provite such a feature intentionally at present. Only the area to open a file is the application icon in the Dock or Finder since this is the standard behavior on a document-based macOS application.

Oh, I didn't know that standard behavior. It mostly solves my problem. Thanks!

**Comment by 2bits** [comment]
To open a file on Drop using CoT, the easiest thing **when I have CoT and a Finder open**
1. in CoT use the File -> Open hotkeys `CTRL+O`
2. you'll see a File Open Dialog box
3. Drop the file into it.
4. press `Return`

Advantages
- The FOD is really big
- It doesn't matter what the FOD shows before drop.

**Reply by ADTC** [reply]
Wow! Coming from Windows, I expected the dropped file to be moved to the current folder in FOD, but in Mac, the FOD switches to the parent folder of the dropped file. What an unexpected difference!

Still, not an ideal solution because you have to switch to CotEditor and open the FOD first, and then you still have to press Enter (or Return as Apple calls it). I think (given the repo author's extreme resistance to implementing a proper drop-to-open), the faster alternative is still the Dock icon as you don't need steps 1, 2 or 4.
